src/callback_tracking.jl

```diff
@@ -206,27 +206,40 @@ function _setup_reverse_callbacks(cb::Union{ContinuousCallback, DiscreteCallback
 
     cb.save_positions == [1, 0] && error("save_positions=[1,0] is currently not supported.")
 
+    # event times
+    times = if typeof(cb) <: DiscreteCallback
+        cb.affect!.event_times
+    else
+        [cb.affect!.event_times; cb.affect_neg!.event_times]
+    end
+
+    # precompute w and wp to generate a tape
+    if cb isa DiscreteCallback
+        w, wp = setup_w_wp(cb, true, nothing, nothing)
+    elseif cb isa ContinuousCallback
+        # compile a tape for pos and neg
+        w, wp = [setup_w_wp(cb, pos_neg, nothing, nothing) for pos_neg in (true, false)]
+    else
+        # cb isa VectorContinuousCallback
+        # compile a tape for pos/neg and all event indices
+        w, wp = [setup_w_wp(cb, pos_neg, event_idx, nothing) for pos_neg in (true, false)
+                 for event_idx in 1:(cb.len)]
+    end
```

Comment:
Here's the part where I am still unsure how to do it in a good way. Should we try to re-use the tools from `adjointdiffcache` to build the tape for the callbacks, or is it better to generate only new ReverseDiff tapes here? (I would hope that if I do the former correctly that we can also support Enzyme)

Reply:
Maybe the first? Though that might be trying too hard.